src/components/Tables/Artists.tsx

```diff
@@ -1,62 +1,139 @@
-import { useMemo, useState } from 'react'
+import { useCallback, useMemo, useState } from 'react'
 import { api } from '~/utils/api'
 import { Artist } from '~/types/data'
 import { HeaderCell } from './components'
 import {
-    ColumnDef,
     SortingState,
     flexRender,
     getCoreRowModel,
     useReactTable,
     getSortedRowModel,
     getFilteredRowModel,
     ColumnFiltersState,
-    getPaginationRowModel
+    getPaginationRowModel,
+    createColumnHelper
 } from '@tanstack/react-table'
-import { Table, Box, Flex } from '@radix-ui/themes'
+import { Table, Box, Flex, Text } from '@radix-ui/themes'
 import Loading from '../Loading'
 import { fuzzyFilter } from './utils/filters'
 import { PaginationButtonGroup } from './components/PaginationButtonGroup'
+import { useRouter } from 'next/router'
+import { TableActionMenu } from './components/TableActionMenu'
+import FadeOutText from '../FadeOutText'
+
+const columnHelper = createColumnHelper<Artist>()
 
 export function ArtistsTable() {
-    const { data, isFetched, isLoading } = api.artist.getAll.useQuery()
+    const { data, isFetched, isLoading, refetch } = api.artist.getAll.useQuery()
+    const router = useRouter()
+    const { mutate: setFeaturedMutation, isSuccess: setFeaturedIsSuccess } =
+        api.artist.setFeatured.useMutation()
+    const { mutate: deleteMutation, isSuccess: deleteMutationIsSuccess } =
+        api.artist.delete.useMutation()
+    const [error, setError] = useState<string | null>(null)
+
+    const handleEditClick = useCallback(
+        async (artist: Artist) => {
+            const params = new URLSearchParams()
+            params.set('id', artist.id)
+            await router.push(
+                {
+                    pathname: '/admin/edit-artist',
+                    query: params.toString()
+                },
+                undefined,
+                { shallow: true }
+            )
+        },
+        [router]
+    )
 
-    const columns = useMemo<ColumnDef<Artist>[]>(
+    const handleToggleFeatured = useCallback(
+        (artist: Artist) => {
+            setFeaturedMutation(
+                { id: artist.id },
+                {
+                    onSuccess: () => {
+                        setError(null)
+                        void refetch()
+                    },
+                    onError: (e) => {
+                        setError(
+                            'Setting featured artist failed. Please try again later.'
+                        )
+                        console.error(e)
+                    }
+                }
+            )
+        },
+        [setFeaturedMutation, refetch]
+    )
+
+    const handleDelete = useCallback(
+        (artist: Artist) => {
+            deleteMutation(
+                { id: artist.id },
+                {
+                    onSuccess: () => {
+                        setError(null)
+                        void refetch()
```

Comment:
I don't think we need `void` here.

Reply:
yes, we do. Try removing it ant then run npx next build. It will fail.